### DISCUSSION TITLE
Firebase SignalR Equalivent?

### DISCUSSION THREAD

**Discussion by longtimedeveloper** [firebase/firebase-js-sdk#5379] [Q&A] [answered]
What is the Firebase SignalR Equalivent?

Thank you for your guidance and congratulations on version 9!

**Comment by jhuleatt** [comment, marked as answer]
Hi @longtimedeveloper! The two Firebase databases, [Realtime Database](https://firebase.google.com/docs/database) and [Cloud Firestore](https://firebase.google.com/docs/firestore), include real-time functionality built-in. In terms of deciding which one to use, take a look at the ["Choose a database" guide](https://firebase.google.com/docs/firestore/rtdb-vs-firestore).

If you'd like to see what the client code looks like for subscribing to realtime events, here are the docs for [Realtime Database listeners](https://firebase.google.com/docs/database/web/read-and-write#web_value_events) and [Cloud Firestore listeners](https://firebase.google.com/docs/firestore/query-data/listen).

I'm not too familiar with SignalR, but if it also includes push notification functionality, you can take a look at [Firebase Cloud Messaging for web](https://firebase.google.com/docs/cloud-messaging/js/client).

**Reply by longtimedeveloper** [reply]
@jhuleatt thank you very much for your outstanding guidance.  Exactly what this new Firebase developer needed.  

SignalR is pub/sub on a grand scale.  I use it for games, now that I'm rewriting them in Flutter, I figured I needed to go with Firebase.

Best to you, Karl
